autogpt_platform/backend/backend/util/cache.py

```diff
@@ -0,0 +1,21 @@
+import threading
+from functools import wraps
+from typing import Callable, TypeVar
+
+T = TypeVar("T")
+R = TypeVar("R")
+
+
+def thread_cached_property(func: Callable[[T], R]) -> property:
```

Comment:
This is different to @cached_property in the sense that the cache scope is on the thread level, not the process level.

Reply:
nice